### PR TITLE
✨ Source Mixpanel: make engage stream really incremental

### LINKED ISSUE #40583: [source-mixpanale] Unable to sync high volume of engage - The stream is not really incremental
### Connector Name

mixpanel

### Connector Version

v3.1.4

### What step the error happened?

During the sync

### Relevant information

The stream engage is not really incremental and doesn't really take into account the start_date to retrieve the data.

That means if we have 1 million data globally, the connector will go through all the data and only filter record according to the start_date even if we want only the last day of data.
Also due to the API rate limiting of Mixpanel, we can retrieve only 1000 records per 60s, means that we need to wait 278 hours each sync only to add the last day of data.

That is due to the non utilization of a clause where in the request : https://github.com/airbytehq/airbyte/blob/4f6c29a9ca04b0c69a9d614fb344abf69ffc6333/airbyte-integrations/connectors/source-mixpanel/source_mixpanel/manifest.yaml#L164-L200

If we can include a clause where in the query like that (exemple for the low code builder):
```
      retriever:
        type: SimpleRetriever
        requester:
          $ref: '#/definitions/base_requester'
          path: /api/query/engage
          http_method: GET
          request_parameters:
            where: properties["$last_seen"] > "{{ stream_interval.start_time }}"
            project_id: '{{ config[''projectId''] }}'
```

We could direct filter on the last_seen property and reduce greatly the incremental sync and even the first sync due to the start_date.

I'm not able to make it completly work in the UI builder due to the session_id token. I'm unable to get the session_token from the last response and send it to the new pagination one.

Possibly related to https://github.com/airbytehq/airbyte/issues/29035 even if this issue was not specificaly evocated.

I'm ok if you guide me to make a PR on this issue.

### Relevant log output

_No response_

### Contribute

- [X] Yes, I want to contribute

### PR DESCRIPTION
**Pull request by descampsk**
## What

Fix https://github.com/airbytehq/airbyte/issues/40583

## How

This PR add a where parameters for the engage stream to be able to retrieve only records according to the start_date or the incremental start_time. This improves greatly the sync time because it does not retrieve all the data and then filter on the start_date.

## Review guide

1. `airbyte-integrations/connectors/source-mixpanel/source_mixpanel/components.py` : Add EngagesHttpRequester with the clause where
2. `airbyte-integrations/connectors/source-mixpanel/source_mixpanel/manifest.yaml`: Update the manifest to use EngagesHttpRequester for the engage stream

## User Impact
Improved performance for the engage stream

## Can this PR be safely reverted and rolled back?

- [x] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Comment by natikgadzhi** [comment]
Wow, thank you @descampsk! Running CI, will merge once things pass.

**Comment by natikgadzhi** [comment]
The code _looks_ like it should work, but I wonder if the incremental behavior is better done with a low-code component instead of a custom component. @ChristoGrab can you take a look and review?

**Comment by descampsk** [comment]
I tried to do it also with the yaml file but without success due to the `MixpanelHttpRequester` but maybe there is still a way.

I'm also currently making some tests about the page_size which set to 1000, to allow the user to override this value.

E.g: I tried with 40 000 and it seems that it works. Therefore it increases the sync sync by a factor 40 :)

Do you perfer an other PR on this subject or shoud I use this one?

**Comment by ChristoGrab** [comment]
@descampsk Thanks for the solid contribution, and apologies for the delay in taking a look at this! I left one small review comment, but otherwise ran some tests and all LGTM 👍

@natikgadzhi Given how reliant this connector generally is on custom components (specifically with the custom requester), I'm inclined to say we accept the change as it stands and create a ticket to investigate where we're at with the CDK being able to handle these needs innately. There were quite a few Mixpanel line items in the spreadsheet tracking low-code feature requests and I'm not sure how far along we've gotten in addressing them

File: airbyte-integrations/connectors/source-mixpanel/source_mixpanel/components.py [hunk 1]
```diff
@@ -102,6 +102,22 @@ def get_request_params(
...
+    def get_request_params(
+        self,
+        *,
+        stream_state: Optional[StreamState] = None,
+        stream_slice: Optional[StreamSlice] = None,
+        next_page_token: Optional[Mapping[str, Any]] = None,
+    ) -> MutableMapping[str, Any]:
+        params = super().get_request_params(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token)
+        if "start_time" in stream_slice:
+            params["where"] = f'properties["$last_seen"] > "{stream_slice["start_time"]}"'
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-mixpanel/source_mixpanel/components.py:115]:
```suggestion
            params["where"] = f'properties["$last_seen"] >= "{stream_slice["start_time"]}"'
```

File: airbyte-integrations/connectors/source-mixpanel/source_mixpanel/components.py [hunk 2]
```diff
@@ -102,6 +102,22 @@ def get_request_params(
...
+        elif "start_date" in self.config:
+            params["where"] = f'properties["$last_seen"] > "{self.config["start_date"]}"'
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-mixpanel/source_mixpanel/components.py:117]:
```suggestion
            params["where"] = f'properties["$last_seen"] >= "{self.config["start_date"]}"'
```

**Review by ChristoGrab — APPROVED** [review]
🚢

**Comment by descampsk** [comment]
Sorry for my delay, I was on vacation this week :) 

Thanks for the merge!

Regarding the page size, to increase the speed of the sync, I build a custom image based on this to set the page limit to 40 000 and it works pretty well.

Do you want a PR on this subject as well @ChristoGrab ?

**Comment by natikgadzhi** [comment]
PR is definitely welcome ;-)

**Pull request merged**